### ISSUE TITLE
Inconsistent FilterComboBox whenClosed() result

### ISSUE THREAD

**Issue by stefnotch** [elix/elix#202]
When toggling the filter combo box with the `v` icon, the previous result will simply be repeated. I would have expected it to be "no user interaction, return undefined"

**Steps that reproduce the issue**
1. Use the following little script
```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <title>Elix FilterComboBox</title>
    <script type="module" src="https://unpkg.com/elix@15.0.0/define/FilterComboBox.js"></script>
  </head>

  <body role="main">
    Open the dev tools.
	Type A and then click on the little v icon.
  
    <div class="demo padded">
      Fruit:
      <elix-filter-combo-box aria-label="Fruits" id="mySearchBox">
        <div>Acai</div>
        <div>Akee</div>
        <div>Apple</div>
        <div>Apricot</div>
        <div>Avocado</div>
      </elix-filter-combo-box>
    </div>
	
	<script>
		mySearchBox.addEventListener("open", (ev) => {
			ev.target.whenClosed().then(result => {console.log(result)});	
		});
	</script>
  </body>
</html>
```
2. Type in some search text and interact with the component.
3. Click on Apple.
4. Click on the `v` icon a few times
5. Open it and click outside of the component
6. Click on the `v` icon a few times

**Expected behavior**
I expected it to return undefined if I click on the `v` icon.

**Comment by stefnotch** [comment]
This also happens with the `.detail` property of the `"close"` event. I tried out the `whenClosed` variant before checking the close event, because the documentation doesn't really tell you that this `.detail` property exists.

**Comment by JanMiksovsky** [comment]
Sorry for the slow response, will take a look as soon as I can.

**Comment by JanMiksovsky** [comment]
Thanks for your patience. I had time to dig into this today.

There appear to be two parts to this issue.

First problem: the relevant mixin, OpenCloseMixin, was resetting the closeResult to undefined whenever a component's `open` method was called. The idea is that, once a component is opened, its closeResult is not defined until the component is closed.

That's correct but insufficient. In this case, the FilterComboBox toggle button is calling the `toggle` method, which skirted the resetting of closeResult to undefined.

I've just checked in a fix for this first problem on #main. The fix resets closeResult to undefined whenever a component's opened state changes to true — regardless of how the component was opened. In particular, the closeResult will now be reset when the `toggle` method opens a combo box.

Second problem: what's an appropriate closeResult when the user closes the component by clicking the toggle?

If the user picked an item, the combo box returns that item via the closeResult. But there are a variety of other possible closeResult values for various UI actions that can close a popup, including: canceling with the Escape key, canceling with the F4 key (a standard popup close key on Windows), and canceling on window blur.

To the extent that closeResult communicates _why_ a popup was closed, it feels like the same goal should apply to closing a popup by clicking the toggle again. For example, we could define a new close result: `{ canceled: "toggle" }`.

(In retrospect, I wish that the closeResult for an item being selected had been something like `{ selected: <item> }`, to indicate both why the combo box was closed and what was selected. But that would be a breaking change now, and I'd prefer to avoid that.)

Before defining a new closeResult for the toggle case, I wanted to see if you had thoughts about it.

**Comment by stefnotch** [comment]
Glad to hear that the problem is now fixed! 
I don't really have a strong opinion as to what the result is, as long as I can work with it.

**Comment by JanMiksovsky** [comment]
Okay. I've gone ahead with a change to report `{ canceled: "toggle" }` in this case.

**Comment by JanMiksovsky** [comment]
This has now been released as Elix 15.0.1.
